### ISSUE TITLE
Macbook pro retina display fonts are pixelated (HiDPI)

### ISSUE THREAD

**Issue by xor-gate** [Brewtarget/brewtarget#259]
I know Qt supports HiDPI and why are we not building releases with HiDPI (since 5.6) ?
The font looks awefull on Macbook pro retina, probably also on Ubuntu powered Dell laptop with HiDPI.

See:
http://doc.qt.io/qt-5/highdpi.html
https://wiki.archlinux.org/index.php/HiDPI#Qt_5

**Comment by rocketman768** [comment]
I didn't see anything in those articles about how to get the "native" scaling from OSX. I found the following on [blog.qt.io](http://blog.qt.io/blog/2013/04/25/retina-display-support-for-mac-os-ios-and-x11/)

>High DPI mode is controlled by the following keys in the Info.Plist file:

    <key>NSPrincipalClass</key>
    <string>NSApplication</string>
    <key>NSHighResolutionCapable</key>
    <string>True</string>

**Comment by rocketman768** [comment]
I'm wondering if you can edit the plist in your installation and see how it looks.

**Comment by mikfire** [comment]
Speaking for myself only.

I have no clue how to enable the builds for hidpmi.

I have no clue what versions of anything we claim to support. I would
prefer to avoid, where possible, enabling new features if it makes
reasonably recent platforms not work.

But. Mostly. It's that I don't have such a display. Until I do, the lack of
support is unlikely bother me. I also don't want to support something I
literally cannot see.

Mik

**Comment by xor-gate** [comment]
I poke around which Qt version with 2.3.0 is shipped:
```
Jerrys-MacBook-Pro:MacOS jerry$ otool -L Brewtarget
Brewtarget:
	@rpath/QtPrintSupport.framework/Versions/5/QtPrintSupport (compatibility version 5.5.0, current version 5.5.1)
	@rpath/QtQml.framework/Versions/5/QtQml (compatibility version 5.5.0, current version 5.5.1)
	@rpath/QtSql.framework/Versions/5/QtSql (compatibility version 5.5.0, current version 5.5.1)
	@rpath/QtXml.framework/Versions/5/QtXml (compatibility version 5.5.0, current version 5.5.1)
	@rpath/QtWebKitWidgets.framework/Versions/5/QtWebKitWidgets (compatibility version 5.5.0, current version 5.5.1)
	@rpath/QtMultimedia.framework/Versions/5/QtMultimedia (compatibility version 5.5.0, current version 5.5.1)
	@rpath/QtWidgets.framework/Versions/5/QtWidgets (compatibility version 5.5.0, current version 5.5.1)
	@rpath/QtWebKit.framework/Versions/5/QtWebKit (compatibility version 5.5.0, current version 5.5.1)
	@rpath/QtNetwork.framework/Versions/5/QtNetwork (compatibility version 5.5.0, current version 5.5.1)
	@rpath/QtGui.framework/Versions/5/QtGui (compatibility version 5.5.0, current version 5.5.1)
	@rpath/QtCore.framework/Versions/5/QtCore (compatibility version 5.5.0, current version 5.5.1)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 120.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1213.0.0)
```

**Comment by xor-gate** [comment]
@rocketman768 I have tried to update the plist but no difference.

**Comment by rocketman768** [comment]
I would like to see this working. Looks crappy on my work mac. I can poke around and see if there's a way to enable highres mode.

**Comment by xor-gate** [comment]
If there needs some testing to be done, just poke here.

**Comment by kapinga** [comment]
I have played around with this some. On Qt 5.6 and later (I tested on 5.8, but the blog post above indicates it should work starting with 5.6.

One line of code is needed to activate HiDPI awareness. In main() in main.cpp, add
`QGuiApplication::setAttribute( Qt::AA_EnableHighDpiScaling );`
before `app` is initialized.
This can also be done by setting the environment variable `QT_AUTO_SCREEN_SCALE_FACTOR` to 1 before running brewtarget.exe (e.g. for testing).

I will note that there's a few UI issues that don't scale nicely in HiDPI mode (even though it's inifinitely better than nothing). In particular, the text on the RangedSliders gets cut off and probably should be changed to 12 pt Arial Bold, not 14 pt Arial Black.

The biggest change though is needing Qt 5.6+. TBH, I have no idea what the decision-making process is or what it involves to upgrade the packaged version of Qt.

![brewtarget hidpi](https://cloud.githubusercontent.com/assets/1765623/23581744/58b8aa2c-00e8-11e7-8397-879422f8f965.PNG)

**Comment by jawsdaws** [comment]
I've made some changes to help with this problem over in [https://github.com/jawsdaws/brewtarget/tree/HiDPI](https://github.com/jawsdaws/brewtarget/tree/HiDPI)  Activating HiDPI awareness is unreliable for me, so I've been using the DPI number to calculate the size of things that need changes.  I only have Linux to test on.  It would be awesome if anyone is willing and able to test that branchon Mac or Windows. :-)

**Comment by matty0ung** [comment]
There is still some more work to do on this, but hopefully https://github.com/Brewtarget/brewtarget/pull/502 has made things a bit better.  (I have a high DPI display on my Linux laptop, but I hope the fixes that work there also benefit Mac users with highdpi...)
As mentioned above, just activating the "work with high DPI" switch is, unfortunately, an unsatisfactory solution.  (On Linux for instance, it does nothing at all as far as I can tell.)  So there is a bit more work to make things appear at sensible sizes on all resolution displays.  Fortunately it's all very doable, so it's mostly a case of finding all the places where things are currently showing poorly on high DPI displays and applying similar fixes to the ones in the patch above.

**Comment by mattiasmaahl** [comment]
fixed in PR #502 , closing
